### DISCUSSION TITLE
Using magic types for list of literal

### DISCUSSION THREAD

**Discussion by cbusillo** [odoo-ide/pycharm-odoo#478] [General]
Hi!  Is there a way to use the TypeDict of "odoo.values.model" as a list of fields for a Literal?  I find myself defining lists of fields for things and I would love a way to validate them with the linter.  I make way too many typos.  I'm giving an example of somewhere I would use it, and the pattern I see in my head.  The alternative I found was using product._fields['default_code'].name.  I didn't like that.

    @api.model
    def _check_missing_data(self, product: "odoo.model.product_base"):
        min_image_size = 50
        required_fields: Literal[list["odoo.values.product_base"]] = [
            "default_code",
            "name",
            "sales_description",
            "standard_price",
            "list_price",
            "qty_available",
            "bin",
            "manufacturerr",
        ]

        missing_fields = [field for field in required_fields if not product[field]]

        if not product.images:
            missing_fields.append("images")

        for image in product.images:
            if image.image_1920_file_size_kb < min_image_size:
                missing_fields.append(
                    f"Image ({image.index}) too small ({image.image_1920_file_size_kb}kB < {min_image_size}kB)"
                )

        return missing_fields

**Comment by trinhanhngoc** [comment]
Hello @cbusillo ,

This feature is already in my backlog to implement. See https://github.com/odoo-ide/pycharm-odoo/issues/452#issuecomment-1906144344.

**Reply by cbusillo** [reply]
Awesome.  Sorry I had no idea what to search for.  I had even read that issue and didn't realize it applied.  I will patiently await it and read change-logs.  There are quite a few places this would have helped me catch bugs in the past.  I'm sure there is a better pattern I can use, but I haven't figured it out yet.

Again, thank you for the plugin, its amazing.
